src/examples/forest.ts

```diff
@@ -0,0 +1,171 @@
+import {
+    Armature,
+    CostFunction,
+    GeneratorInstance,
+    GuidingVectors,
+    Light,
+    Material,
+    Node,
+    Point,
+    Renderer,
+    RGBColor,
+    Shape
+} from '../calder';
+
+import { range } from 'lodash';
+
+// tslint:disable-next-line:import-name
+import Bezier = require('bezier-js');
+
+// Create the renderer
+const renderer: Renderer = new Renderer({
+    width: 800,
+    height: 600,
+    maxLights: 2,
+    ambientLightColor: RGBColor.fromRGB(90, 90, 90),
+    backgroundColor: RGBColor.fromHex('#FFDDFF')
+});
+
+// Create light sources for the renderer
+const light1: Light = Light.create({
+    position: { x: 10, y: 10, z: 10 },
+    color: RGBColor.fromHex('#FFFFFF'),
+    strength: 200
+});
+
+// Add lights to the renderer
+renderer.addLight(light1);
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Step 1: create geometry
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Setup leaf
+const leafColor = RGBColor.fromRGB(204, 255, 204);
+const leafSphere = Shape.sphere(Material.create({ color: leafColor, shininess: 100 }));
+
+// Setup branch
+const branchColor = RGBColor.fromRGB(102, 76.5, 76.5);
+const branchShape = Shape.cylinder(Material.create({ color: branchColor, shininess: 1 }));
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Step 2: create armature
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+const bone = Armature.define((root: Node) => {
+    root.createPoint('base', { x: 0, y: 0, z: 0 });
+    root.createPoint('mid', { x: 0, y: 0.5, z: 0 });
+    root.createPoint('tip', { x: 0, y: 1, z: 0 });
+    root.createPoint('handle', { x: 1, y: 0, z: 0 });
+});
+
+const treeGen = Armature.generator();
+treeGen
+    .define('forest', (_: Point, instance: GeneratorInstance) => {
```

Comment:
The notable difference between the `forest` and `generation` files (in terms of the generator) is that the forest has  this rule to generate a bunch of branch spawn points within a square on the ground.